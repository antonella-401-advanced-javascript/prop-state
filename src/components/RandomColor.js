import React, { Component } from 'react';
import ColorDisplay from '../colors/ColorDisplay';

export default class RandomColor extends Component {
  state = {
    backgroundColor: 'black'
  };

  selectColor = () => {
    const colors = ['red', 'green', 'black', 'yellow', 'purple', 'grey', 'pink', 'brown', 'blue'];
    const random = Math.floor(Math.random() * colors.length);
    this.setState({ backgroundColor: colors[random] });
  };

  componentDidMount() {
    this.selectColor();
    setInterval(this.selectColor, 1000);
  }

  render() {
    const { backgroundColor } = this.state;
  
    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
      </>
    );
  }
}