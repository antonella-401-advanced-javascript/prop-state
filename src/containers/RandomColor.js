import React, { Component } from 'react';
import ColorDisplay from '../colors/ColorDisplay';

export default class RandomColor extends Component {
  state = {
    backgroundColor: 'black'
  };

  selectColor = () => {
    const colors = ['red', 'green', 'black', 'yellow', 'purple', 'grey', 'pink', 'brown', 'blue', 'lavender', 'thistle', 'magenta', 'darkorchid', 'indigo', 'mediumpurple', 'lightcoral', 'peachpuff', 'darkseagreen', 'seagreen', 'darkturquoise', 'darkcyan', 'steelblue', 'hotpink', 'mistyrose', 'maroon'];
    const random = Math.floor(Math.random() * colors.length);
    this.setState({ backgroundColor: colors[random] });
  };

  componentDidMount() {
    this.selectColor();
    setInterval(this.selectColor, 1000);
  }

  render() {
    const { backgroundColor } = this.state;
  
    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
      </>
    );
  }
}