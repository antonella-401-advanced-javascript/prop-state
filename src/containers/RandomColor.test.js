import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

Enzyme.configure({ adapter: new Adapter() });
describe('Random Color component', () => {
  it('renders Random Color', () => {
    const wrapper = shallow(<RandomColor />);
    wrapper.setState({ backgroundColor: 'pink' });
    expect(wrapper).toMatchSnapshot();
  });
});