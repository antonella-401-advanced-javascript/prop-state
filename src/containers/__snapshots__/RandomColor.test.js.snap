// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Color component renders Random Color 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="pink"
  />
</Fragment>
`;
